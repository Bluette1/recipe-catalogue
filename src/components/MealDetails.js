import React from 'react';
import PropTypes from 'prop-types';
import QueryString from 'query-string';
import '../css/MealsDetails.css';

const MealDetails = props => {
  const { location: { search } } = props;
  const parsedParams = QueryString.parse(search);
  const {
    img, t, c, i, st, y, ins,
  } = parsedParams;
  return (
    <div>
      <div className="header">
        <ul className="mealslist-heading">
          <li>
            Recipe catalogue
          </li>
          <a href="/"><button type="button">Back</button></a>
        </ul>
      </div>
      {' '}
      <div className="image details-pg">
        <div
          className="image-area details-pg"
          style={{
            backgroundImage: `url(${img})`,
            backgroundRepeat: 'no-repeat',
            backgroundPosition: '0% 0%',
            backgroundSize: 'cover',
          }}
        />
      </div>
      <div className="meal-row details-pg">
        <div className="title-category">
          <p className="category">{c}</p>
          <p className="area category">{st}</p>
          <h4 className="title">{t}</h4>
        </div>
        <div>
          <h4>
            Ingredients:
            {i}
          </h4>
        </div>
      </div>
      <div className="meal-row ins">
        <div>
          <div className="ins-heading">
            <h4 className="instructions">Instructions:</h4>
            <a href={y}><button type="button">Youtube Video</button></a>
          </div>
          <p className="body-ins">{ins}</p>
        </div>
      </div>
    </div>
  );
};

MealDetails.propTypes = {
  location: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default MealDetails;
