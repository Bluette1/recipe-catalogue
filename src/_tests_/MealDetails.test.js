import React from 'react';
import {
  render, cleanup,
} from '@testing-library/react';

import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore, { history } from '../store';
import MealDetails from '../components/MealDetails';

configure({ adapter: new Adapter() });

afterEach(cleanup);

const store = configureStore();

const MealWithStore = () => (
  <Provider store={store}>
    <React.StrictMode>
      <ConnectedRouter history={history}>
        <MealDetails location={{ search: 'img=imgUrl&t=title&i=ingredients&y=youtubeLink&ins=instructuctions' }} />
      </ConnectedRouter>
    </React.StrictMode>
  </Provider>
);

it('renders the Meal component', () => {
  const rendered = render(<MealWithStore />);
  const mealTitle = rendered.getByText('Instructions:');
  expect(mealTitle).toBeInTheDocument();

  const mealCategory = rendered.getByText('Ingredients: ingredients');
  expect(mealCategory).toBeInTheDocument();

  const streetArea = rendered.getByText('title');
  expect(streetArea).toBeInTheDocument();
  expect(rendered).toMatchSnapshot();
});

it('renders the Meal Details component with the link to the home [age]', () => {
  const rendered = mount(<MealWithStore />);
  const link = rendered.find('a').get(0);

  const homepage = '/';
  expect(link.props.href).toBe(homepage);
  expect(rendered).toMatchSnapshot();
});
it('renders the Meal Details component with the link to the youtube video', () => {
  const rendered = mount(<MealWithStore />);
  const link = rendered.find('a').get(1);

  const youtubeUrl = 'youtubeLink';
  expect(link.props.href).toBe(youtubeUrl);
  expect(rendered).toMatchSnapshot();
});

it('renders the Meal component with the correct buttons', () => {
  const rendered = mount(<MealWithStore />);
  const youtubeBtn = rendered.find('[data-testid="youtubeLink"]');
  expect(youtubeBtn.text()).toBe('Youtube Video');

  const backBtn = rendered.find('[data-testid="homeLink"]');
  expect(backBtn.text()).toBe('Back');
  expect(rendered).toMatchSnapshot();
});
