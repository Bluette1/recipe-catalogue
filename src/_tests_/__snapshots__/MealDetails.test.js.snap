// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Meal Details component with the link to the home [age] 1`] = `ReactWrapper {}`;

exports[`renders the Meal Details component with the link to the youtube video 1`] = `ReactWrapper {}`;

exports[`renders the Meal component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="details-bg"
      >
        <div
          class="header"
        >
          <ul
            class="mealslist-heading"
          >
            <li>
              Recipe catalogue
            </li>
            <a
              href="/"
            >
              <button
                data-testid="homeLink"
                type="button"
              >
                Back
              </button>
            </a>
          </ul>
        </div>
        <div
          class="image-area details-pg"
          style="background-image: url(imgUrl); background-repeat: no-repeat; background-position: 0% 0%; background-size: cover;"
        />
        <div
          class="details-pg"
        >
          <div
            class="title-categ"
          >
            <h4
              class="titl"
            >
              title
            </h4>
          </div>
          <div
            class="ingredients"
          >
            <h4>
              Ingredients: 
              ingredients
            </h4>
          </div>
        </div>
        <div
          class="ins"
        >
          <div>
            <div
              class="ins-heading"
            >
              <h4
                class="instructions"
              >
                Instructions:
              </h4>
              <a
                href="youtubeLink"
              >
                <button
                  data-testid="youtubeLink"
                  type="button"
                >
                  Youtube Video
                </button>
              </a>
            </div>
            <p
              class="body-ins"
            >
              instructions
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="details-bg"
    >
      <div
        class="header"
      >
        <ul
          class="mealslist-heading"
        >
          <li>
            Recipe catalogue
          </li>
          <a
            href="/"
          >
            <button
              data-testid="homeLink"
              type="button"
            >
              Back
            </button>
          </a>
        </ul>
      </div>
      <div
        class="image-area details-pg"
        style="background-image: url(imgUrl); background-repeat: no-repeat; background-position: 0% 0%; background-size: cover;"
      />
      <div
        class="details-pg"
      >
        <div
          class="title-categ"
        >
          <h4
            class="titl"
          >
            title
          </h4>
        </div>
        <div
          class="ingredients"
        >
          <h4>
            Ingredients: 
            ingredients
          </h4>
        </div>
      </div>
      <div
        class="ins"
      >
        <div>
          <div
            class="ins-heading"
          >
            <h4
              class="instructions"
            >
              Instructions:
            </h4>
            <a
              href="youtubeLink"
            >
              <button
                data-testid="youtubeLink"
                type="button"
              >
                Youtube Video
              </button>
            </a>
          </div>
          <p
            class="body-ins"
          >
            instructions
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the Meal component with the correct buttons 1`] = `ReactWrapper {}`;
